sound/soc/sof/intel/hda-dai.c

```diff
@@ -164,7 +164,8 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 		snd_hdac_ext_bus_link_set_stream_id(hlink, stream_tag);
 
 	/* set the hdac_stream in the codec dai */
-	snd_soc_dai_set_stream(codec_dai, hdac_stream(hext_stream), substream->stream);
+	if (ops->codec_dai_set_stream)
```

Comment:
is this op meant to be optional @plbossart

Reply:
yes, it's not needed for SoundWire/DMIC/SSP. It's only required for HDaudio codecs.